### PR TITLE
GCU test for bgp speaker

### PR DESCRIPTION
**Pull request by wen587**
### Description of PR

[Done]Wait for yang model https://github.com/Azure/sonic-buildimage/pull/10082 
and create-only fix to merge.
Summary: Testcase of bgp speaker config for generic updater apply-patch
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
End to End test support for Generic Updater apply-patch
This PR is to verify the usage of 'config apply-patch' works on bgp speaker config

#### How did you do it?
Add bgp speaker config to dut and check if config change as expected

#### How did you verify/test it?
Run test of sonic-mgmt/tests/generic_config_updater/test_bgp_speaker.py on KVM

### CODE REVIEW

**Review by ganglyu — DISMISSED** [review]

File: tests/generic_config_updater/test_bgp_speaker.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    elif ip_version == "v6":
+        lo_intf_ip = lo_ipv6
+        vlan_intf_ip_range = ip_rangev6
+        bgp_speaker = BGPSPEAKER_V6
+        dummy_ip_range = DUMMY_IP_RANGE_V6
+    else:
+        pytest.fail("Invalid ip version!")
+
+    bgp_speaker_config_cleanup(duthost)
+    bgp_speaker_tc1_add_config(duthost, lo_intf_ip, vlan_intf_ip_range, bgp_speaker)
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_bgp_speaker.py:195]: >bgp_speaker_tc1_add_config [](http://example.com/codeflow?start=4&length=26)

So you test with ipv4 only, and then ipv6 only. Should you test add ipv6 and ipv6 config in the same JsonPatch?

Comment (wen587) [reply tests/generic_config_updater/test_bgp_speaker.py:195]: I guess they are the same. Do you think testing them together is better?

File: tests/generic_config_updater/test_bgp_speaker.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    bgp_speaker_tc1_add_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range)
+    bgp_speaker_tc1_rm_dummy_ip_range(duthost, bgp_speaker, dummy_ip_range)
```
Comment (ghooo) [inline tests/generic_config_updater/test_bgp_speaker.py:197]: Can we add a test for changing the `src_address` switching from loopback0 to loobpack1?

Also, we would like to test adding a new VLAN and also a new Loopback is that covered under VLAN/Loopback testing? (i think so, just double checking).

Comment (wen587) [reply tests/generic_config_updater/test_bgp_speaker.py]: Added src_address change to a dummy ip.

> Also, we would like to test adding a new VLAN and also a new Loopback is that covered under VLAN/Loopback testing? (i think so, just double checking).

Yes. That is covered.

File: tests/generic_config_updater/test_bgp_speaker.py (hunk 1 above)
Comment (wen587) [reply tests/generic_config_updater/test_bgp_speaker.py]: combined.

**Review by ghooo — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**